### ISSUE TITLE
Add support for unified push

### ISSUE THREAD

**Issue by ne20002** [friendica/friendica#11698]
### Is the feature request related to a problem? Please describe.

As of version 19 Tusky will support unified push (see https://unifiedpush.org). It would be great if Friendica would support this.

### Describe the feature you'd like

Have a setting to enable use of unified push
Have a configuration to choose unfied push provider

With Nextpush setting up a own unified push provider is easy. It already supports matrix protocol / synapse server and matrix client Element is also adding support. It would be handy to have this for Friendica with Tusky as well. A hope, other foss apps will also implement it.

### Additional context

**Comment by foss-** [comment]
Fedilab also supports unified push. Having an alternative to Google Push is a huge win.

**Comment by kr428** [comment]
Adding myself here. Currently, fedilab (as maybe the most feature-complete ActivityPub client) doesn't at all seem to be able to display Friendica notifications as they fully focussed on UnifiedPush so essentially using Friendica with fedilab hasn't working (push?) notifications at the moment. Though they also offer sort of a pull notification process, I never managed to get that to work.

**Comment by xundeenergie** [comment]
There ist https://ntfy.sh as a very well done unified-push and notification server.
I run it for matrix, and it works very well!!

**Comment by sjehuda** [comment]
Did you consider XMPP PubSub instead?

XMPP PubSub exists for over a couple of decades, and it delivers well.

https://xmpp.org/extensions/xep-0060.xml

**Comment by kr428** [comment]
> Did you consider XMPP PubSub instead?
> 
> XMPP PubSub exists for over a couple of decades, and it delivers well.

Not sure though whether this is still relevant. Also, this, all along with #13057, was/is more about supporting a convenient notification experience on mobile devices, in my cases specifically Android ones, and I wanted to keep the amount of apps and clients that are about to receive push notifications limited. For most of these cases and specifically here, it seems UnifiedPush is what most of the open / privacy-focused apps do support out of the box. For XMPP PubSub it would boil down to needing dedicated clients on mobile devices supporting this.

Personally, I'm fine with e-mail notifications at the moment. UP would be cool but there are other aspects on my wishlist that seem more important related to Friendica.

**Comment by sjehuda** [comment]
> For XMPP PubSub it would boil down to needing dedicated clients on
> mobile devices supporting this.

This statement is not accurate.

It is possible to add support of XMPP client capabilities.

**Comment by kr428** [comment]
> This statement is not accurate.
> 
> It is possible to add support of XMPP client capabilities.

Would this work out of the box with something like tusky, fedilab, ... ?

**Comment by sjehuda** [comment]
> Would this work out of the box with something like tusky, fedilab,
> ... ?
> 

I suppose that it would not, due to lack of support of XMPP.

Nevertheless, the developers of Bonfire and Pleroma intend to support
XMPP for publishing, so XMPP might very well be even more accessible in
future.

**Comment by kr428** [comment]
> Would this work out of the box with something like tusky, fedilab,
>  ... ?
> I suppose that it would not, due to lack of support of XMPP.
> 

Yeah, but that was the main objective of this issue - make sure Friendica supports the current means of notifications implemented with the /existing/ apps out there. That's why I meant XMPP needs dedicated client support first which might not be a sure thing as most of the existing apps don't really care about even explicitely supporting current Friendica notifications out of the box either and resort to using Mastodon client API which isn't likely to use XMPP I guess.🙂

**Comment by sjehuda** [comment]
It seems that too many people think that XMPP is a platform which is
only for messaging, even though it is the first ever to be a federated
publishing platform.

I sense that, due to that misrealization, it be good to have an XEP of
Activity Streams, so it would intrigue people to explore about XMPP and
publishing.

https://github.com/friendica/friendica/issues/15306

**Comment by annando** [comment]
@sjehuda this issue here is about integrating *existing* apps with Friendica that already use this specific push. Your point can be discussed in a separate issue.

**Comment by sjehuda** [comment]
Fine. Sorry.

I figured that it is related, because that mechanism, as I realize it,
already exists in XMPP (XEP-0060).

I will create a separate issue, and refer to this, if I would be sure
of my statement.
